README.md
SIMPLE HOTEL MANAGEMENT SYSTEM IN C++

DEVELOPED BY SHIVJI KUMAR (2019BCS-058) AND PARAG PODDAR (2019BCS-058)


history.txt
Wed Oct 21 09:13:52 2020
	Check In 	301	MVR	shivaji kumar	delhi - 110041	8295850524
Wed Oct 21 09:36:14 2020
	Check In 	302	MVR	shiv	delhi-110071	8765432109
Tue Nov 03 12:50:25 2020
	Check In 	107	OBR	parag	bus stand	11110000110


room.h
#include<bits/stdc++.h>
using namespace std;

namespace rooms{
	
	class room{
		 private:
		 	int roomNo;
		 	int price;
		 	string type;
		    

		public:
			// constructor
			room(){
				this->roomNo = 0;
				this->price = 0 ;
				this->type = "none";
			}
			
			
			void setRoomNo(int no){
				 this->roomNo = no;
			}
			void setRoomPrice(int price){
				this->price = price;
			}
			void setRoomType(string s){
				this->type = s;
			}
			int getRoomNo(){
				return roomNo;
			}
			string getRoomType(){
				return this->type;
			}
			int getRoomPrice(){
				return price;
			}
			
			
			
			//to make it abstract function make a pure virtual function;
			virtual void fun1()=0;   // now room is a abstract class and no objects can be formed
			                          // of room class.
	
			//destructor 
			~room(){
              delete &roomNo;
              delete &price ;
			}
	};
	
	// mountain view rooms class
	class mountainViewRoom:public room{
		 
		 public:
		 	mountainViewRoom(int no){
		 		room::setRoomNo(no);
		 		room::setRoomType("MVR");
			 }
			 
			 //hiding the fun1 of parent class
			 void fun1(){
			 	cout<< "hiding the fun1 of parent class";
			 }
	};
	
	// 1 bed room class
	class oneBedRoom: public room{
		 
		 public:
		 		oneBedRoom(int no){
			 		room::setRoomNo(no);
			 		room::setRoomType("OBR");
				 }
				 void fun1(){
				 	cout<< "hiding the fun1 of parent class";
				 }
	};
	
	// 3 bed room class
	class threeBedRoom: public room{

		 public:
		 		threeBedRoom(int no){
			 		room::setRoomNo(no);
			 		room::setRoomType("TBR");
				 }
				 void fun1(){
				 	cout<< "hiding the fun1 of parent class";
				 }
	};
	
	
}


history.h
#include<bits/stdc++.h>
#include <ctime>
using namespace std;


namespace History{
	
	class history{
		// all function are public beacause they all will be used outside of the class
		public:
		   void	checkInHistory(int rNo,string type, string name,string address,unsigned long long mobile);
		   void checkOutHistory(int rNo,string type, string name,string address,unsigned long long mobile,int amount);
		   void showHistory();  
	};
	
	//insert data in history.txt every time a 
	// check in take palace 
	void history::checkInHistory(int rNo,string type, string name,string address,unsigned long long mobile){
		time_t tt;
		struct tm *ti;
		time(&tt);
		ti = localtime(&tt);
		
		fstream f;
		f.open("history.txt",ios::app);
		f<<asctime(ti)<<"\tCheck In \t"<<rNo<<"\t"<<type<<"\t"<<name<<"\t"<<address<<"\t"<<mobile<<endl;
		f.close();
	}
	
	//insert data in history.txt every time a 
	// check out take palace 
	void history::checkOutHistory(int rNo,string type, string name,string address,unsigned long long mobile,int amount){
		time_t tt;
		struct tm *ti;
		time(&tt);
		ti = localtime(&tt);
		
		fstream f;
		f.open("history.txt",ios::app);
		f<<asctime(ti)<<"\tCheck Out \t"<<rNo<<"\t"<<type<<"\t"<<name<<"\t"<<address<<"\t"<<mobile<<"\t"<<amount<<endl;
		f.close();
	}
	
	//show history for admin
	void history::showHistory(){
		fstream f("history.txt",ios::in);
		char date[100],otherDetail[2000];
		
		while(!f.eof())
		{
			f.getline(date,100);
			f.getline(otherDetail,2000);
			cout<<date<<"\t"<<otherDetail<<endl;
		}
		
		cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
		getch();
		system("cls");
        return ;
		}
	
}



adminFun.h
#include<bits/stdc++.h>
#include<conio.h>
#include <string>
#include "history.h"

using namespace std;
using namespace History;

namespace adminFun{
	
	class action{
		 
	    history h;   // object of history class in history.h header file
	    bool checkUnoccupancy(int rNO);
		void changeRoomStatus();
		void changeRoomPrice();
		
		//check for unoccupied rooms
		void AllUnoccupied();
		//check for occupied rooms
		void AllOccupied();
		//check for all not for service rooms
		void AllNotForServiceRooms();
				
		public:
		void mainMenu();
	};
	
	// function for check unoccupancy of a room
	bool action::checkUnoccupancy(int rNo){
			fstream f;
			string status;
			f.open(to_string(rNo)+".txt",ios::in);
			f>>status;
			
			if(status != "occupied")	return true;
			else  return false;
	    }
	
	
//	  for change room 
//	  status of any unoccupied room,
//    if the room is not prepared or
//	  book for vip
//	  or any other reason

    void action::changeRoomStatus(){
    	  system("cls");
		 // room and stayer person details
		 int roomno;
		 fstream f;
		 label2: 	 		
		 cout << "Enter room no from 101 to 110 or 201 to 205 or 301 to 305 whose status you whant to change : ";
	   	 cin>>roomno;
	   	        if(checkUnoccupancy(roomno)){
	   	        	char status[15];
	   	        	char state[15],roomType[10];
	   	        	int price;
			 		cout <<"what is status want to write (* less then 15 character) : ";
			 		cin.ignore();
			 		cin>>status;
			 		f.open(to_string(roomno)+".txt",ios::in);
			 		f>>state; f>>roomType; f>>price;
			 		f.close();
			 		
			 		f.open(to_string(roomno)+".txt",ios::out);
					f<<status<<endl<<roomType<<endl<<price<<endl;
					f<<"noone"<<endl<<"nowhere"<<endl<<9876543210<<endl;
					f.close();
					
					cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
					getch();
					system("cls");
					mainMenu();			 		
				}
				else {
				 	cout << "this room is currently booked please select any other room no\n ";
				 	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				 	getch();
				 	system("cls");
				 	goto label2;
				}
	}
	
//	for change the price 
//	of the room of any unoccupied room 

    void action::changeRoomPrice(){
    	  system("cls");
		 // room and stayer person details
		 int roomno;
		 fstream f;
		 label3: 	 		
		 cout << "Enter room no from 101 to 110 or 201 to 205 or 301 to 305 : ";
	   	 cin>>roomno;
	   	        if(checkUnoccupancy(roomno)){
	   	        	char state[15],roomType[10];
	   	        	int price;
			 		cout <<"what is New price : ";
			 		cin>>price;
			 		f.open(to_string(roomno)+".txt",ios::in);
			 		f>>state; f>>roomType;
			 		f.close();
			 		
			 		f.open(to_string(roomno)+".txt",ios::out);
					f<<state<<endl<<roomType<<endl<<price<<endl;
					f<<"noone"<<endl<<"nowhere"<<endl<<9876543210<<endl;
					f.close();	
					
					
					cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
					getch();
					system("cls");
					mainMenu();		 		
				}
				else {
				 	cout << "this room is currently booked please select any other room no\n ";
				 	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				 	getch();
				 	system("cls");
				 	goto label3;
				}
	
	}
	
	  // function to get all unoccupied rooms
		 
	void action::AllUnoccupied(){
		  cout<<"\n\n\t\t\t\tAll unoccupied rooms are as follows\n\n\n";
		  int i=101;
		  while(i<306){
		  	 string status,type,rType;
			 int price;
		  	 fstream f;
		  	 f.open(to_string(i)+".txt",ios::in);
		  	 f>>status; f>>type; f>>price;
		  	 if(type=="MVR") rType="Mountain View Room";
		  	 else if(type == "OBR") rType="One Bed Room";
		  	 else rType = "Three Bed Room";
		  	 if(status=="unoccupied"){
		  	 	cout<<"\t\t"+rType +"\t\t"<<i<<"\t\t"<<price<<"\n\n";
		     }
		  	
		  	
		  	if(i==110) i=201;
			else if(i==205) i=301;
			else i++;
		  }
		  
		    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
			getch();
			system("cls"); 
			mainMenu();
	}	 
	
	
	//function to get all occupied room
	
	void action::AllOccupied(){
		cout<<"\n\n\t\t\t\tAll unoccupied rooms are as follows\n\n\n";
		  int i=101;
		  while(i<306){
		  	 string status,type,rType;
		  	 char name[50], address[800];
		  	 unsigned long long mobile;
			 int price;
		  	 fstream f;
		  	 f.open(to_string(i)+".txt",ios::in);
		  	 f>>status; f>>type; f>>price;
		  	 f.ignore();
		  	 f.getline(name,50); f.getline(address,800);
		  	 f>>mobile;
		  	 if(type=="MVR") rType="Mountain View Room";
		  	 else if(type == "OBR") rType="One Bed Room";
		  	 else rType = "Three Bed Room";
		  	 if(status=="occupied"){
		  	 	cout<<"\t\tRoomType : "+rType +"\t\tRoomNo : "<<i<<"\t\tRent : "<<price<<endl;
		  	 	cout<<"\t\tName : "<<name<<"\t\t\taddress : "<<address<<"\t\t mobile : "<<mobile<<"\n\n\n";
		     }
		  	
		  	
		  	if(i==110) i=201;
			else if(i==205) i=301;
			else i++;
		  }
		  
		    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
			getch();
			system("cls"); 
			mainMenu();
	}
		 
    // function to show all rooms which are not
    //available for booking due to any reason
		 
    void action::AllNotForServiceRooms(){
    	cout<<"\n\n\t\t\t\tAll NOt For Service rooms are as follows\n\n\n";
		  int i=101;
		  while(i<306){
		  	 string status,type,rType;
			 int price;
		  	 fstream f;
		  	 f.open(to_string(i)+".txt",ios::in);
		  	 f>>status; f>>type; f>>price;
		  	 if(type=="MVR") rType="Mountain View Room";
		  	 else if(type == "OBR") rType="One Bed Room";
		  	 else rType = "Three Bed Room";
		  	 if(status !="unoccupied" && status !="occupied"){
		  	 	cout<<"\t\t"+rType +"\t\t"<<i<<"\t\t"<<price<<"\n\n";
		     }
		  	
		  	
		  	if(i==110) i=201;
			else if(i==205) i=301;
			else i++;
		  }
		  
		    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
			getch();
			system("cls"); 
			mainMenu();
	}
		 //mainmenu  for admin
		 //the options in switch
		 //case 
		 
void action::mainMenu()
	{
	
		int choice;
		label1:
		while(choice!=5)
		{
	
			  system("cls");
			cout<<"\n\t\t\t\t*************************";
			cout<<"\n\t\t\t\t SIMPLE HOTEL MANAGEMENT ";
			cout<<"\n\t\t\t\t      * MAIN MENU *";
			cout<<"\n\t\t\t\t*************************";
			cout<<"\n\n\n\t\t\t1.Change Room Status of a unoccupid room";
			cout<<"\n\t\t\t2.Change Room Price of a unoccupid room ";
			cout<<"\n\t\t\t3.All Unoccupied Rooms";
			cout<<"\n\t\t\t4.All Occupied Room Details";
			cout<<"\n\t\t\t5.All Rooms which are not available for service";
			cout<<"\n\t\t\t6.Show the history";
			cout<<"\n\t\t\t7.Exit";
			cout<<"\n\n\t\t\tEnter Your Choice: ";
			cin>>choice;
	
			switch(choice)
			{
			
			case 1:	changeRoomStatus();
					break;
			
			case 2: changeRoomPrice();
					break;
			
			case 3: AllUnoccupied();
			break;
			
			case 4:	AllOccupied();
					break;
			
			case 5: AllNotForServiceRooms();
			        break;
				  
			case 6: h.showHistory();
			        break;
			        
			case 7: return ;
			
			default:
				{
				cout<<"\n\n\t\t\tWrong choice.....!!!";
				cout<<"\n\t\t\tPress any key to   continue....!!";
				getch();
				goto label1;
				}
			
			}
	
	    }
	
	}
	
	
	
	//end of namespace
}


receptionFun.h
#include<bits/stdc++.h>
#include<conio.h>
#include<string>
#include "room.h"
#include "customer.h"
#include "history.h"
using namespace std;
using namespace rooms;
using namespace customers;
using namespace History;

namespace functions{
	
	class action{
		history h;
		//booking of room 
	    bool checkUnoccupancy(int rNO);
		void bookARoom();
		void saveData(string status,mountainViewRoom mvr , customer cust);
		void saveData2(string status,oneBedRoom ovr , customer cust);
		void saveData3(string status,threeBedRoom tvr , customer cust);
		//checkout
		bool checkOccuancy(int rNo);
		void checkout();
		//check for unoccupied rooms
		void AllUnoccupied();
		//check for occupied rooms
		void AllOccupied();
		
		
		public:
		void mainMenu();
	};
	
	    //check unoccupancy of the room 
	    //with perticular room no
	
		bool action::checkUnoccupancy(int rNo){
			fstream f;
			string status;
			f.open(to_string(rNo)+".txt",ios::in);
			f>>status;
			
			if(status == "unoccupied")	return true;
			else  return false;
	    }
	    
	
	    void action::saveData(string status,mountainViewRoom mvr , customer cust){
				int price ; char state[15],roomType[10]; 
				fstream f;
				f.open(to_string(mvr.getRoomNo())+".txt",ios::in);
				f.getline(state,15);
				f.getline(roomType,10);
				f>>price;
				f.close();
				
				f.open(to_string(mvr.getRoomNo())+".txt",ios::out);
				f<<status<<endl<<mvr.getRoomType()<<endl<<price<<endl;
				f<<cust.getName()<<endl<<cust.getAddress()<<endl<<cust.getMobile()<<endl;
				f.close();
				
				//save to history of all customer data .....
				h.checkInHistory(mvr.getRoomNo(),mvr.getRoomType(),cust.getName(),cust.getAddress(),cust.getMobile());
				
				cout<<"\n Room is booked...!!!\n";
		    	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				getch();
		        mainMenu();        
		
	    }
	
	    void action::saveData2(string status,oneBedRoom ovr , customer cust){
				int price ; char state[15],roomType[10]; 
				fstream f;
				f.open(to_string(ovr.getRoomNo())+".txt",ios::in);
				f.getline(state,15);
				f.getline(roomType,10);
				f>>price;
				f.close();
				
				f.open(to_string(ovr.getRoomNo())+".txt",ios::out);
				f<<status<<endl<<ovr.getRoomType()<<endl<<price<<endl;
				f<<cust.getName()<<endl<<cust.getAddress()<<endl<<cust.getMobile()<<endl;
				f.close();
				
				//save to history of all customer data .....
				h.checkInHistory(ovr.getRoomNo(),ovr.getRoomType(),cust.getName(),cust.getAddress(),cust.getMobile());
				
				cout<<"\n Room is booked...!!!\n";
		    	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				getch();
		        mainMenu(); 
		}
	
		void action::saveData3(string status,threeBedRoom tvr , customer cust){
				int price ; char state[15],roomType[10]; 
				fstream f;
				f.open(to_string(tvr.getRoomNo())+".txt",ios::in);
				f.getline(state,15);
				f.getline(roomType,10);
				f>>price;
				f.close();
				
				f.open(to_string(tvr.getRoomNo())+".txt",ios::out);
				f<<status<<endl<<tvr.getRoomType()<<endl<<price<<endl;
				f<<cust.getName()<<endl<<cust.getAddress()<<endl<<cust.getMobile()<<endl;
				f.close();
				
				//save to history of all customer data .....
				h.checkInHistory(tvr.getRoomNo(),tvr.getRoomType(),cust.getName(),cust.getAddress(),cust.getMobile());
				
				
				cout<<"\n Room is booked...!!!\n";
		    	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				getch();
		        mainMenu(); 
		}
	
	void action::bookARoom(){
		 system("cls");
		 int choise;
		 // room and stayer person details
		 char status[15], roomType[10], name[50],address[800];
		 int price,roomno;
		 unsigned long long mobile;
		 
		 fstream f;
		 cout<<"\nwhich type of room you want to book\nPress 1 for MOUNTAIN VIEW ROOMS\n";
		 cout<<"Press 2 for ONE BED ROOMS \nPress 3 for THREE BED ROOMS"<<endl;
		 
		 cin>>choise;
		 
		 switch(choise){
		 	case 1:	
			    label2: 	 		
		 		cout << "Enter room no from 301 to 305 : ";
				cin>>roomno;
		 		if(roomno<301 || roomno>305){
		 			 cout<< "please enter a valid room no";
		 			 cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
		 			 getch();
		 			 system("cls");
		 			 goto label2;
				 }
		 		  
		 		
		 	    // checking avability of that room no ;
				 if(checkUnoccupancy(roomno)){
					mountainViewRoom mvr1(roomno);   // creating object of a m.v. room 
			 		customer  cust;   //creating object of customer
			 		
			 		cout<<"\n Enter Customer Detalis";
					cout<<"\n ----------------------"<<endl;
					cin.ignore() ; // to remove input buffer 
					cout<<" Name: ";
					cin.getline(name,50);
					cout<<" Address: ";
					cin.getline(address,800);
					cout<<" Phone No: ";
					cin>>mobile;
					
					cust.SetPerson(name,address,mobile); // seting person detail in customer object 
					saveData("occupied",mvr1,cust);
				}
				else {
				 	cout << "this room is not available for booking please select any other room no\n ";
				 	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				 	getch();
				 	system("cls");
				 	goto label2;
				}
				break;
				 
			case 2:
				label3: 
		 		cout << "Enter room no from 101 to 110 : ";
				cin>>roomno;
		 		if(roomno<101 || roomno>110){
		 			 cout<< "please enter a valid room no";
		 			 cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
		 			 getch();
		 			 system("cls");
		 			 goto label3;
				 }
		 		  
		 		
		 	    // checking avability of that room no ;
				 if(checkUnoccupancy(roomno)){
					oneBedRoom ovr(roomno);   // creating object of a m.v. room 
			 		customer  cust;   //creating object of customer
			 		
			 		cout<<"\n Enter Customer Detalis";
					cout<<"\n ----------------------"<<endl;
					cin.ignore() ; // to remove input buffer 
					cout<<" Name: ";
					cin.getline(name,50);
					cout<<" Address: ";
					cin.getline(address,800);
					cout<<" Phone No: ";
					cin>>mobile;
					
					cust.SetPerson(name,address,mobile); // seting person detail in customer object 
					saveData2("occupied",ovr,cust);
				 }else {
				 	cout << "this room is not available for booking please select any other room no\n ";
				 	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				 	getch();
				 	system("cls");
				 	goto label3;
				 } 	
				 break;
				 
			case 3:
			    label4:
		 		cout << "Enter room no from 201 to 205 : ";cin>>roomno;
		 		if(roomno<201 || roomno>205){
		 			 cout<< "please enter a valid room no";
		 			 cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
		 			 getch();
		 			 system("cls");
		 			 goto label4;
				 }
		 		  
		 		
		 	    // checking avability of that room no ;
				 if(checkUnoccupancy(roomno)){
					threeBedRoom tvr(roomno);   // creating object of a m.v. room 
			 		customer  cust;   //creating object of customer
			 		
			 		cout<<"\n Enter Customer Detalis";
					cout<<"\n ----------------------"<<endl;
					cin.ignore() ; // to remove input buffer 
					cout<<" Name: ";
					cin.getline(name,50);
					cout<<" Address: ";
					cin.getline(address,800);
					cout<<" Phone No: ";
					cin>>mobile;
					
					cust.SetPerson(name,address,mobile); // seting person detail in customer object 
					saveData3("occupied",tvr,cust);
				 }else {
				 	cout << "this room is not available for booking please select any other room no\n ";
				 	cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				 	getch();
				 	system("cls");
				 	goto label4;
				 }
				 break;
			
			default :
			    cout<<"enter a valid input ";
			    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
				getch();
				system("cls");
				mainMenu();
		 }
	}

  //function for checkout 
  // first we check
  // that the room is really occupied 
  //or not, then is it is
  //occupied then we perform checkout 	
	
	
bool action::checkOccuancy(int rNo){
       	fstream f;
		string status;
		f.open(to_string(rNo)+".txt",ios::in);
		f>>status;
		
		if(status == "occupied"){
			return true;
		}  
		else{
			return false;
		} 
}

// function for checkout 

void action::checkout(){
	        	int roomno;
	        		label2: 
		 		cout << "\t\t\t\t\tEnter room no from 101 to 110 or 201 t0 205 or 301 to 305 : ";
			    cin>>roomno;
				
				if(checkOccuancy(roomno)){
						int price ; char status[15],roomType[10],name[50],address[800]; 
						int amount;
						unsigned long long mobile;
						fstream f;
						f.open(to_string(roomno)+".txt",ios::in);
						f.getline(status,15);
						f.getline(roomType,10);
						f>>price;
						f.getline(name,50); f.getline(address,800);
						f>>mobile;
						f.close();
						cout <<"Enter amount the customer pays (no of days * "<<price<<") : ";
						cin>>amount;
						f.open(to_string(roomno)+".txt",ios::out);
						f<<"unoccupied"<<endl<<roomType<<endl<<price<<endl;
						f<<"noone"<<endl<<"nowhere"<<endl<<9876543210<<endl;
						f.close();
						
						//save to history of all customer data .....
		                h.checkOutHistory(roomno,roomType,name,address,mobile,amount);
		
		
						cout<< "\t\t\t\t\tRoom no "<<roomno<<" is unoccupied now the person in this room have checkout just.";
						cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
						getch();
						system("cls");
						mainMenu();
						
				}else{
					cout<< "this room is already empty ";
					cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
					getch();
					system("cls");
					goto label2;
				}
				 
		 }
		 
		 // function to get all unoccupied rooms
		 
	void action::AllUnoccupied(){
		  cout<<"\n\n\t\t\t\tAll unoccupied rooms are as follows\n\n\n";
		  int i=101;
		  while(i<306){
		  	 string status,type,rType;
			 int price;
		  	 fstream f;
		  	 f.open(to_string(i)+".txt",ios::in);
		  	 f>>status; f>>type; f>>price;
		  	 if(type=="MVR") rType="Mountain View Room";
		  	 else if(type == "OBR") rType="One Bed Room";
		  	 else rType = "Three Bed Room";
		  	 if(status=="unoccupied"){
		  	 	cout<<"\t\t"+rType +"\t\t"<<i<<"\t\t"<<price<<"\n\n";
		     }
		  	
		  	
		  	if(i==110) i=201;
			else if(i==205) i=301;
			else i++;
		  }
		  
		    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
			getch();
			system("cls"); 
			mainMenu();
	}	 
	
	
	//function to get all occupied room
	
	void action::AllOccupied(){
		cout<<"\n\n\t\t\t\tAll unoccupied rooms are as follows\n\n\n";
		  int i=101;
		  while(i<306){
		  	 string status,type,rType;
		  	 char name[50], address[800];
		  	 unsigned long long mobile;
			 int price;
		  	 fstream f;
		  	 f.open(to_string(i)+".txt",ios::in);
		  	 f>>status; f>>type; f>>price;
		  	 f.ignore();
		  	 f.getline(name,50); f.getline(address,800);
		  	 f>>mobile;
		  	 if(type=="MVR") rType="Mountain View Room";
		  	 else if(type == "OBR") rType="One Bed Room";
		  	 else rType = "Three Bed Room";
		  	 if(status=="occupied"){
		  	 	cout<<"\t\tRoomType : "+rType +"\t\tRoomNo : "<<i<<"\t\tRent : "<<price<<endl;
		  	 	cout<<"\t\tName : "<<name<<"\t\t\taddress : "<<address<<"\t\t mobile : "<<mobile<<"\n\n\n";
		     }
		  	
		  	
		  	if(i==110) i=201;
			else if(i==205) i=301;
			else i++;
		  }
		  
		    cout<<"\n\n\n\n\n\n\n\t\t\t\t\tPress any key to continue....!!";
			getch();
			system("cls"); 
			mainMenu();
	}
		 
		 
		 //mainmenu for reception
		 //the options in switch
		 //case 
		 
void action::mainMenu()
	{
	
		int choice;
		label1:
		while(choice!=5)
		{
	
			  system("cls");
			cout<<"\n\t\t\t\t*************************";
			cout<<"\n\t\t\t\t SIMPLE HOTEL MANAGEMENT ";
			cout<<"\n\t\t\t\t      * MAIN MENU *";
			cout<<"\n\t\t\t\t*************************";
			cout<<"\n\n\n\t\t\t1.Book A Room";
			cout<<"\n\t\t\t2.Check Out ";
			cout<<"\n\t\t\t3.All Unoccupied Rooms";
			cout<<"\n\t\t\t4.All Occupied Room Details";
			cout<<"\n\t\t\t5.Exit";
			cout<<"\n\n\t\t\tEnter Your Choice: ";
			cin>>choice;
	
			switch(choice)
			{
			
			case 1:	bookARoom();
			break;
			
			case 2: checkout();
			break;
			
			case 3: AllUnoccupied();
			break;
			
			case 4:	AllOccupied();
			break;
			
			case 5: return ;
			
			default:
				{
				cout<<"\n\n\t\t\tWrong choice.....!!!";
				cout<<"\n\t\t\tPress any key to   continue....!!";
				getch();
				goto label1;
				}
			
			}
	
	    }
	
	}
		
	
}



